Source/Core/Core/Boot/AncastTypes.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+
+// Magic number
+#define ANCAST_MAGIC 0xEFA282D9
+
+// The location in memory where PPC ancast images are booted from
+#define ESPRESSO_ANCAST_LOCATION_PHYS 0x01330000
+#define ESPRESSO_ANCAST_LOCATION_VIRT 0x81330000
+
+// The image type
+enum AncastImageType
+{
+    ANCAST_IMAGE_TYPE_ESPRESSO_WIIU = 0x11,
+    ANCAST_IMAGE_TYPE_ESPRESSO_WII  = 0x13,
+    ANCAST_IMAGE_TYPE_STARBUCK_NAND = 0x21,
+    ANCAST_IMAGE_TYPE_STARBUCK_SD   = 0x22,
+};
+
+// The console type of the image
+enum AncastConsoleType
+{
+    ANCAST_CONSOLE_TYPE_DEV     = 0x01,
+    ANCAST_CONSOLE_TYPE_RETAIL  = 0x02,
+};
+
+// Start of each header
+struct AncastHeaderBlock
+{
+    u32 magic;
+    u32 unknown;
+    u32 header_block_size;
+    u8 padding[0x14];
+};
+static_assert(sizeof(AncastHeaderBlock) == 0x20, "AncastHeaderBlock has the wrong size");
+
+// Signature block for type 1
+struct AncastSignatureBlockv1
+{
+    u32 signature_type;
+    u8 signature[0x38];
+    u8 padding[0x44];
+};
+
+// Signature block for type 2
+struct AncastSignatureBlockv2
+{
+    u32 signature_type;
+    u8 signature[0x100];
+    u8 padding[0x7c];
+};
+
+// General info about the image
+struct AncastInfoBlock
+{
+    u32 unknown;
+    u32 image_type;
+    u32 console_type;
+    u32 body_size;
+    u8 body_hash[0x14];
```

Comment:
`std::array<u8, 0x14>` (this makes comparisons nicer in DolReader)

Reply:
Is `sizeof(std::array<u8, N>) == N` guaranteed? Otherwise this would break the `sizeof`'s for the ancast header.

Reply:
Technically it's not guaranteed by the standard -- a conforming implementation could put stuff after the array data in std::array. However it's hard to imagine why any implementation would do that and `sizeof(std::array<char, N>) == N` holds for all common standard library implementations (libstdc++, libc++, MSVC STL).